src/harness/testcases/WINNF_FT_S_PCR_testcase.py

```diff
@@ -342,3 +423,384 @@ def test_WINNF_FT_S_PCR_1(self, config_filename):
+        "cbsdIds": cbsd_ids,
+        "palIds": pal_ids,
+        "providedContour": config['userClaimedPpaContour']
+    }
+
+    # Trigger PPA Creation to SAS UUT.
+    self.assertPpaCreationFailure(ppa_creation_request)
+
+  def generate_PCR_7_default_config(self, filename):
+    """Generate the WinnForum configuration for PCR 7."""
+    # File path where SAS UUT claimed ppa boundary generated in PCR.1 test
+    pcr_1_test_config_file_path = os.path.join('testcases', 'configs',
+                                               'test_WINNF_FT_S_PCR_1',
+                                               'default.config')
+    sas_uut_claimed_ppa_boundary_file_path = getSasUutClaimedPpaBoundaryFilePath(
+      'default.config')
+
+    # Load SAS UUT claimed ppa boundary and check if any error while retrieving
+    # SAS UUT claimed ppa boundary generated in PCR.1 test.
+    try:
+      with open(sas_uut_claimed_ppa_boundary_file_path, 'r') as claimed_ppa_file:
+        user_claimed_ppa_contour = json.load(claimed_ppa_file)
+    except IOError:
+      raise RuntimeError('ConfigError:There is an error in reading path:%s \n\n'
+                         % sas_uut_claimed_ppa_boundary_file_path)
+
+    # Shrink the user claimed ppa boundary by 2 kilometer.
+    user_claimed_ppa_contour_shapely = utils.ToShapely(
+        user_claimed_ppa_contour['features'][0]['geometry']).buffer(-2e-2)
+    user_claimed_ppa_contour_geometry = json.loads(utils.ToGeoJson(
+        user_claimed_ppa_contour_shapely))
+
+    # Create ppa_record where user claimed PPA contour will be replaced.
+    overlap_ppa_record = json.load(
+      open(os.path.join('testcases', 'testdata', 'ppa_record_0.json')))
+
+    # Update the user_claimed ppa contour geometry required for overlaps ppa.
+    overlap_ppa_record['zone'] = user_claimed_ppa_contour_geometry
+
+    # Create the actual config.
+    config = {
+        'configPCR_1': pcr_1_test_config_file_path,
+        'overlapPpaRecord': overlap_ppa_record,
+        'userClaimedPpaContour': user_claimed_ppa_contour_geometry
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_PCR_7_default_config)
+  def test_WINNF_FT_S_PCR_7(self, config_filename):
+    """Overlapping PPA Boundaries.
+
+    Checks SAS UUT shall doesnt create PPA zone within the service area.
+    Checks SAS UUT shall confirm a valid PPA boundary claimed by the PAL holder,
+    composed of one or more adjacent Census Tracts was overlapped by another
+    PPA zone.
+    """
+    # Load the Config file.
+    config = loadConfig(config_filename)
+
+    # Load the test_WINNF_FT_S_PCR_1 config. All other inputs must be identical
+    # to those used in the corresponding configuration of PCR.1.
+    pcr_1_test_config = loadConfig(config['configPCR_1'])
+
+    # Set the PAL frequency.
+    pal_low_frequency = 3570000000
+    pal_high_frequency = 3580000000
+
+    user_id = pcr_1_test_config['registrationRequests'][0]['userId']
```

Comment:
I think a lot of this should be done in the config generation function, not in the test case itself. Please take another look at the spec and let me know if you have any questions.